### PR TITLE
options: don't report errors on help value for OPT_FLAG and OPT_COLOR

### PR DESCRIPTION
**Pull request by pavelxdd**
The same idea as in 3723e61 but for OPT_FLAG now.

Before the change:
-----
````
mpv --border=help
Invalid parameter for border flag: help
Valid values are:
    yes
    no
    (passing nothing)
Error parsing option border (option parameter could not be parsed)
Setting commandline option --border=help failed.

Exiting... (Fatal error)
````

````
mpv --border=sdjfkhlsdfh
Invalid parameter for border flag: sdjfkhlsdfh
Valid values are:
    yes
    no
    (passing nothing)
Error parsing option border (option parameter could not be parsed)
Setting commandline option --border=sdjfkhlsdfh failed.

Exiting... (Fatal error)
````

After the change:
-----
````
mpv --border=help
Valid values for border flag are:
    yes
    no
    (passing nothing)
````

````
mpv --border=sdjfkhlsdfh
Invalid parameter for border flag: sdjfkhlsdfh
Valid values are:
    yes
    no
    (passing nothing)
Error parsing option border (option parameter could not be parsed)
Setting commandline option --border=sdjfkhlsdfh failed.

Exiting... (Fatal error)
````

### CODE REVIEW

**Comment by pavelxdd** [comment]
Added a commit implementing the same behavior for `OPT_COLOR`.

File: options/m_option.c [hunk 1]
```diff
@@ -135,13 +135,18 @@ static int parse_flag(struct mp_log *log, const m_option_t *opt,
...
-                BSTR_P(name), BSTR_P(param));
-    mp_info(log, "Valid values are:\n");
+    bool is_help = bstr_equals0(param, "help");
+    if (!is_help) {
+        mp_fatal(log, "Invalid parameter for %.*s flag: %.*s\n",
+                 BSTR_P(name), BSTR_P(param));
+        mp_info(log, "Valid values are:\n");
+    } else {
+        mp_info(log, "Valid values for %.*s flag are:\n", BSTR_P(name));
+    }
```
Comment (wm4) [inline options/m_option.c]: Better reorder if/else body, so it's `if (cond)` instead of `if (!cond)`

File: options/m_option.c [hunk 2]
```diff
@@ -1792,10 +1796,11 @@ static int parse_color(struct mp_log *log, const m_option_t *opt,
 error:
     mp_err(log, "Option %.*s: invalid color: '%.*s'\n",
            BSTR_P(name), BSTR_P(param));
-    mp_err(log, "Valid colors must be in the form #RRGGBB or #AARRGGBB (in hex)\n"
-                "Or in the form 'r/g/b/a', where each component is a value in the\n"
-                "range 0.0-1.0. (Also allowed: 'gray', 'gray/a', 'r/g/b'.\n");
-    return M_OPT_INVALID;
+print_help:
```
Comment (wm4) [inline options/m_option.c]: Would probably be better to have only one label (at the position of the error label), and print the error only if `!is_help`?

**Comment by wm4** [comment]
LGTM, made some minor improvement suggestions.

**Comment by pavelxdd** [comment]
Updated, thanks for suggestions.
Added a 3rd commit for `OPT_GEOMETRY`

Before the change:
-----
````
mpv --geometry=help
Option geometry: invalid geometry: 'help'
Valid format: [W[%][xH[%]]][{+-}X[%]{+-}Y[%]] | [X[%]:Y[%]]
Error parsing option geometry (option parameter could not be parsed)
Setting commandline option --geometry=help failed.

Exiting... (Fatal error)
````

After the change:
-----
````
mpv --geometry=help
Valid format: [W[%][xH[%]]][{+-}X[%]{+-}Y[%]] | [X[%]:Y[%]]
````

````
mpv --geometry=helpdsffds
Option geometry: invalid geometry: 'helpdsffds'
Valid format: [W[%][xH[%]]][{+-}X[%]{+-}Y[%]] | [X[%]:Y[%]]
Error parsing option geometry (option parameter could not be parsed)
Setting commandline option --geometry=helpdsffds failed.

Exiting... (Fatal error)
````

**Comment by jeeb** [comment]
Passes *nix and Windows builds and generally looks good to me.

**Pull request merged**